src/viam/sdk/module/service.hpp

```diff
@@ -87,6 +94,7 @@ class ModuleService : viam::module::v1::ModuleService::Service {
     std::string parent_addr_;
     std::shared_ptr<Server> server_;
     SignalManager signal_manager_;
+    void (*cleanup_function_)();
```

Comment:
Switched to what I think is a function pointer (just a pointer to a function?). The syntax here is pretty foreign to me so let me know if this looks off for a pointer to a function that takes no arguments and returns nothing. Tested with a module that has no cleanup function, one that has a cleanup function that logs something, and one that has a cleanup function that throws an error. Things seem to work as expected.

Reply:
What motivated the switch from `std::function` to a function pointer? I'd much prefer the former, really.

Reply:
The cleanup function is optional and the previous structure was causing error logging in cases where no function was set. The thinking was a ptr allows us to check if the function has been set at all, and only log if `cleanup_function_ != nullptr`. If there is a cleaner way to do this that doesn't involve using a function ptr I'm certainly happy to use that instead!

Reply:
Ah, but a default constructed `std::function` is false-y: https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool. You can't call it, but you can definitely do:

```
if (cleanup_function_)
    cleanup_function_();
```

Reply:
Oh, great! No need to go pointer route then.

Reply:
Ah nice; switched back to a `std::function<void()>`  but left the guard in around the actual call to `cleanup_function_`.

Reply:
Did you push that commit to the PR? I still looks like it is using a bare function pointer.

Reply:
...oops, sorry! Just force pushed.